yb-voyager/cmd/conflictDetectionCache.go

```diff
@@ -85,34 +138,69 @@ func (c *ConflictDetectionCache) RemoveEvents(batch *tgtdb.EventBatch) {
 	}
 }
 
-func (c *ConflictDetectionCache) eventsConfict(event1, event2 *tgtdb.Event) bool {
-	if !(event1.SchemaName == event2.SchemaName && event1.TableName == event2.TableName) {
+func (c *ConflictDetectionCache) eventsConfict(cachedEvent, incomingEvent *tgtdb.Event) bool {
+	if !(cachedEvent.SchemaName == incomingEvent.SchemaName && cachedEvent.TableName == incomingEvent.TableName) {
 		return false
 	}
 
+	maybeQualifiedName := cachedEvent.TableName
+	if cachedEvent.SchemaName != "public" {
+		maybeQualifiedName = fmt.Sprintf("%s.%s", cachedEvent.SchemaName, cachedEvent.TableName)
+	}
+	uniqueKeyColumns := c.tableToUniqueKeyColumns[maybeQualifiedName]
+
 	/*
 		Not checking for value of unique key values conflict in case of export from yb because of inconsistency issues in before values of events provided by yb-cdc
 		TODO(future): Fix this in our debezium voyager plugin
 
 		For now, we just check if the event is from same table then we consider it as a conflict
 	*/
-	if isTargetDBExporter(event2.ExporterRole) {
-		log.Infof("conflict detected for table %s, between event1(vsn=%d) and event2(vsn=%d)", event1.TableName, event1.Vsn, event2.Vsn)
-		return true
-	}
+	if isTargetDBExporter(incomingEvent.ExporterRole) {
+		conflict := false
+		if cachedEvent.Op == "d" {
+			conflict = true
+		} else if cachedEvent.Op == "u" {
+			// if both events are dealing with the same unique key columns then we consider it as a conflict
+			cachedEventCols := lo.Keys(cachedEvent.Fields)
+			incomingEventCols := lo.Keys(incomingEvent.Fields)
+			ukList := lo.Intersect(cachedEventCols, uniqueKeyColumns)
+			if lo.Some(incomingEventCols, ukList) {
+				conflict = true
+			}
+		}
 
-	maybeQualifiedName := event1.TableName
-	if event1.SchemaName != "public" {
-		maybeQualifiedName = fmt.Sprintf("%s.%s", event1.SchemaName, event1.TableName)
+		if conflict {
+			log.Infof("conflict detected for table %s, between event1(vsn=%d) and event2(vsn=%d)", cachedEvent.TableName, cachedEvent.Vsn, incomingEvent.Vsn)
+		}
+		return conflict
 	}
-	uniqueKeyColumns := c.tableToUniqueKeyColumns[maybeQualifiedName]
+
 	for _, column := range uniqueKeyColumns {
-		// if the unique key column value is same, then we have a conflict
-		if *event1.Fields[column] == *event2.Fields[column] {
+		if cachedEvent.BeforeFields[column] == nil || incomingEvent.Fields[column] == nil {
+			return false
+		}
+		if CompareConflictingEventValues(cachedEvent, incomingEvent, column) {
 			log.Infof("conflict detected for table %s, column %s, between value of event1(vsn=%d, colVal=%s) and event2(vsn=%d, colVal=%s)",
-				maybeQualifiedName, column, event1.Vsn, *event1.Fields[column], event2.Vsn, *event2.Fields[column])
+				maybeQualifiedName, column, cachedEvent.Vsn, *cachedEvent.BeforeFields[column], incomingEvent.Vsn, *incomingEvent.Fields[column])
 			return true
 		}
 	}
 	return false
 }
+
+func CompareConflictingEventValues(cachedEvent, incomingEvent *tgtdb.Event, column string) bool {
```

Comment:
Would you need all this complexity if you move the call to `WaitUntilNoConflict()` before the call to `ConvertEvent()`?

Reply:
hmm possible, let me verify that